src/lib.rs

```diff
@@ -348,17 +398,27 @@ fn clear_cloexec(fd: RawFd) -> nix::Result<()> {
 }
 
 /// Attempt to start a new instance of this proxy.
-fn spawn_child(restart_fd: Option<RawFd>) -> io::Result<process::Child> {
+async fn spawn_child(
```

Comment:
Btw this function should probably be unsafe, as callers need to make sure `restart_fd` isn't a fd that would be unsound to pass to the child.

Reply:
I see your point, but since this fd is created by us, we can wrap it appropriately to ensure safety through the type system.